### PR TITLE
ALTER KEYSPACE statement is treated as unrecognized

### LINKED ISSUE #198: ALTER KEYSPACE statement is treated as unrecognized CQLStatement in audit mapping
Hi,

Found a bug which ALTER KEYSPACE statement is treated as unrecognized CQLStatement in audit mapping.

I tried to run below statement in cqlsh:
ALTER KEYSPACE system_traces WITH replication = {'class': 'NetworkTopologyStrategy', 'dc2': '2', 'dc1': '2'}  AND durable_writes = true;

and i got a warn msg in cassandra log:
[Native-Transport-Requests-1] [com.ericsson.bss.cassandra.ecaudit.entry.factory.AuditEntryBuilderFactory] Detected unrecognized CQLStatement in audit mapping"

### PR DESCRIPTION
**Pull request by tommystendahl**
Closes #198

### CODE REVIEW

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/StatementResourceAdapter.java [hunk 1]
```diff
@@ -45,10 +47,15 @@
...
+class StatementResourceAdapter //NOPMD
 {
 
-    private static final String FAILED_TO_RESOLVE_RESOURCE = "Failed to resolve resource";
+    private static final String UNCHECKED = "unchecked";
+	private static final String ARGUMENTS = "arguments";
+	private static final String FAILED_TO_RESOLVE_BASE_TABLE_OF_FUNCTION = "Failed to resolve base table of function";
+	private static final String NAME = "name";
+	private static final String TABLE_NAME = "tableName";
+	private static final String FAILED_TO_RESOLVE_RESOURCE = "Failed to resolve resource";
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/StatementResourceAdapter.java:58]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/StatementResourceAdapter.java [hunk 2]
```diff
@@ -45,10 +47,15 @@
...
 import org.apache.cassandra.schema.TableMetadataRef;
 
-class StatementResourceAdapter
+class StatementResourceAdapter //NOPMD
 {
 
-    private static final String FAILED_TO_RESOLVE_RESOURCE = "Failed to resolve resource";
+    private static final String UNCHECKED = "unchecked";
+	private static final String ARGUMENTS = "arguments";
+	private static final String FAILED_TO_RESOLVE_BASE_TABLE_OF_FUNCTION = "Failed to resolve base table of function";
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/StatementResourceAdapter.java:55]: fixed

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 3]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(USE_PERMISSIONS)
                          .resource(DataResource.table(statement.getAuditLogContext().keyspace, statement.getAuditLogContext().scope));
     }
 
+    private Builder createDropTriggerEntryBuilder(DropTriggerStatement.Raw statement)
+    {
+        QualifiedName name;
+        try
+        {
+            name = (QualifiedName) FieldUtils.readField(statement, "tableName", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:1004]: fixed

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 4]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(ALTER_PERMISSIONS)
                          .resource(DataResource.keyspace(statement.getAuditLogContext().keyspace));
     }
 
+    private Builder createAlterTypeEntryBuilder(AlterTypeStatement.Raw statement)
+    {
+        UTName name;
+        try
+        {
+            name = (UTName) FieldUtils.readField(statement, "name", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:850]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 5]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(CREATE_PERMISSIONS)
                          .resource(DataResource.keyspace(statement.getAuditLogContext().keyspace));
     }
 
+    private Builder createCreateTypeEntryBuilder(CreateTypeStatement.Raw statement)
+    {
+        UTName name;
+        try
+        {
+            name = (UTName) FieldUtils.readField(statement, "name", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:827]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 6]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(DROP_PERMISSIONS)
                          .resource(DataResource.table(statement.getAuditLogContext().keyspace, statement.getAuditLogContext().scope));
     }
 
+    private Builder createDropTableEntryBuilder(DropTableStatement.Raw statement)
+    {
+        QualifiedName name;
+        try
+        {
+            name = (QualifiedName) FieldUtils.readField(statement, "name", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:762]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 7]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(ALTER_PERMISSIONS)
                          .resource(DataResource.table(statement.getAuditLogContext().keyspace, statement.getAuditLogContext().scope));
     }
 
+    private Builder createAlterTableEntryBuilder(AlterTableStatement.Raw statement)
+    {
+        QualifiedName name;
+        try
+        {
+            name = (QualifiedName) FieldUtils.readField(statement, "name", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:739]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 8]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(DROP_PERMISSIONS)
                          .resource(DataResource.keyspace(statement.getAuditLogContext().keyspace));
     }
 
+    private Builder createDropKeyspaceEntryBuilder(DropKeyspaceStatement.Raw statement)
+    {
+        String keyspaceName;
+        try
+        {
+            keyspaceName = (String) FieldUtils.readField(statement, "keyspaceName", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:702]: done

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java [hunk 9]
```diff
@@ -525,146 +555,464 @@ private Builder createSchemaAlteringEntryBuilder(CQLStatement statement)
...
                          .permissions(ALTER_PERMISSIONS)
                          .resource(DataResource.keyspace(statement.getAuditLogContext().keyspace));
     }
 
+    private Builder createAlterKeyspaceEntryBuilder(AlterKeyspaceStatement.Raw statement)
+    {
+        String keyspaceName;
+        try
+        {
+            keyspaceName = (String) FieldUtils.readField(statement, "keyspaceName", true);
```
[the remark this answers is not in the document]
Comment (tommystendahl) [reply ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/factory/AuditEntryBuilderFactory.java:679]: done

**Review by masokol — COMMENTED** [review]
LGTM

**Pull request merged**